e2e/helper.go

```diff
@@ -65,49 +28,22 @@ func NewContainer(image string, privatePort int) *Container {
 
 // Start starts a new docker container on a random port
 func (c *Container) Start(t *testing.T) {
-	cmd := exec.Command("docker", "run", "--rm", "--privileged", "-d", "-P", c.image)
-	output := runCmd(t, cmd)
-	c.container = strings.Trim(output, " \r\n")
+	result := icmd.RunCommand("docker", "run", "--rm", "-d", "-P", c.image).Assert(t, icmd.Success)
+	c.container = strings.Trim(result.Stdout(), " \r\n")
 	time.Sleep(time.Second * 3)
 }
 
 // Stop terminates this container
 func (c *Container) Stop(t *testing.T) {
-	runCmd(t, exec.Command("docker", "stop", c.container))
+	icmd.RunCommand("docker", "stop", c.container).Assert(t, icmd.Success)
 }
 
 // GetAddress returns the host:port this container listens on
 func (c *Container) GetAddress(t *testing.T) string {
 	if c.address != "" {
 		return c.address
 	}
-	cmd := exec.Command("docker", "port", c.container, strconv.Itoa(c.privatePort))
-	output := runCmd(t, cmd)
-	c.address = fmt.Sprintf("127.0.0.1:%v", strings.Trim(strings.Split(output, ":")[1], " \r\n"))
+	result := icmd.RunCommand("docker", "port", c.container, strconv.Itoa(c.privatePort)).Assert(t, icmd.Success)
+	c.address = fmt.Sprintf("127.0.0.1:%v", strings.Trim(strings.Split(result.Stdout(), ":")[1], " \r\n"))
 	return c.address
 }
-
-// AssertCommand runs command, assert it succeeds, return its output
-func AssertCommand(t *testing.T, exe string, args ...string) []byte {
```

Comment:
I feel like it would be nice to keep the `AssertCommand` helper even if we're switching to `icmd`, since we're using it so frequently (and will continue to do so as we add more tests)

```golang
func AssertCommand(t *testing.T, dir string, cmd... string) *icmd.Result {
  return icmd.RunCmd(icmd.Cmd{
    Command: cmd,
    Dir:     dir,
  }).Assert(t, icmd.Success)
}
```

Reply:
I kinda disagree here, the whole purpose of `icmd` is to be composable, if we re-do an helper func for 4 lines used 4 times, I don't see it as important. 

> duplication is far cheaper than the wrong abstraction

Reply:
Don't want to insist more than necessary, especially since it's already been merged, but the abstraction I proposed would preserve composability since we return the Result object. And the pattern is used way more than 4 times, so that's a bit disingenuous IMO.

@ijc Not sure I understand your point. We don't maintain `gotest.tools`, so I think we should be leveraging what it does provide and write the helpers that make sense for our own codebase.

Reply:
> We don't maintain gotest.tools

It is Open Source and accepts contributions.